apps/gnome-web/src/components/ui/admin/gnome-details.tsx

```diff
@@ -0,0 +1,118 @@
+import { useNavigate, useParams } from "react-router-dom";
+import { toast } from "sonner";
+import UserIcon from "@/assets/icons/users-icon.svg";
+import PlaceHolder from "@/assets/images/placeholder.png";
+import { Button } from "@/components/ui/button";
+import { useDistrictStore } from "@/store/useDistrictStore";
+import { useGnomeStore } from "@/store/useGnomeStore";
+
+function useToastNavigate() {
+  const navigate = useNavigate();
+
+  const toastNavigate = (
+    path: string,
+    message: string,
+    type: "success" | "error" | "info" = "info",
+  ) => {
+    switch (type) {
+      case "success":
+        toast.success(message);
+        break;
+      case "error":
+        toast.error(message);
+        break;
+      default:
+        toast(message);
+    }
+    navigate(path);
+  };
+
+  return toastNavigate;
+}
+
+function GnomeDetail() {
```

Comment:
wg. mnie powinno to być w page'u gnoma a nie w osobnym komponencie

komponenty wydzielamy raczej po to aby były reużywalne albo żeby podzielić jakoś logicznie kod, tutaj wyciągnięta jest cała strona i przyznam szczerze że nie widzę w tym jakichś większych plusów